pkg/proxy/nftables/proxier.go

```diff
@@ -723,11 +723,12 @@ func CleanupLeftovers(ctx context.Context) bool {
 
 	for _, family := range []knftables.Family{knftables.IPv4Family, knftables.IPv6Family} {
 		nft, err := knftables.New(family, kubeProxyTable)
-		if err == nil {
-			tx := nft.NewTransaction()
-			tx.Delete(&knftables.Table{})
-			err = nft.Run(ctx, tx)
+		if err != nil {
+			continue
 		}
+		tx := nft.NewTransaction()
+		tx.Delete(&knftables.Table{})
+		err = nft.Run(ctx, tx)
 		if err != nil && !knftables.IsNotFound(err) {
```

Comment:
this was checking before the error returned from `nft, err := knftables.New` and now you directly continue, don't you want to retain the log and set the encounteredError? 
does the `!knftables.IsNotFound(err) {` apply to nft.Run?

Reply:
No, if `knftables.New` fails then either (a) there is no `nft` binary or (b) there is no kernel support for nftables, and so either way _there can't be stale nftables rules to clean up_ so we don't need to log anything.

More specifically, if you run `kube-proxy --proxy-mode iptables` on a host with no `nft` binary, it should not print out an error about there not being any `nft` binary, because you didn't ask it to do anything with `nft`.